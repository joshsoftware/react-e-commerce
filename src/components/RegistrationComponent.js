import React from 'react';
import FormCard from './FormCard';
import ContainerWrapper from './ContainerWrapper';
import RowWrapper from './RowWrapper';
import ColumnWrapper from './ColumnWrapper';
import PropTypes from 'prop-types';
import ImagePreviewContainer from '../containers/ImagePreviewContainer';

const RegistrationComponent = ({ validateData, dispatch, formState }) => {
  const { imageUrl } = formState;

  return (
    <ContainerWrapper
      styleClass={'pt-3 text-center h-100'}
      data={
        <RowWrapper
          data={
            <>
              <ColumnWrapper
                sm={{ size: 6, offset: 3 }}
                data={
                  <FormCard
                    type={'registration'}
                    validateData={validateData}
                    dispatch={dispatch}
                    formState={formState}
                  />
                }
              />
              <ImagePreviewContainer imageUrl={imageUrl} />
            </>
          }
        />
      }
    />
  );
};

export default RegistrationComponent;

RegistrationComponent.propTypes = {
  validateData: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
  formState: PropTypes.object.isRequired
};
