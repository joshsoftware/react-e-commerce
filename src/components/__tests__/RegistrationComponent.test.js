import { render, cleanup } from '@testing-library/react';
import React from 'react';
import RegistrationComponent from '../RegistrationComponent';

afterEach(cleanup);

describe('RegistrationComponent', () => {
  const validateData = () => {};

  const dispatch = () => {};

  const formState = {
    firstname: '',
    lastname: '',
    email: '',
    password: '',
    country: '',
    state: '',
    city: '',
    address: ''
  };

  it('Must return RegistrationComponent', () => {
    const { asFragment } = render(
      <RegistrationComponent
        validateData={validateData}
        dispatch={dispatch}
        formState={formState}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
