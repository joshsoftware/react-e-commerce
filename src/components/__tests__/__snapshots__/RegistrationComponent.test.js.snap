// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationComponent Must return RegistrationComponent 1`] = `
<DocumentFragment>
  <div
    class="pt-3 text-center h-100 container"
  >
    <div
      class="row"
    >
      <div
        class="col-sm-6 offset-sm-3"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <h3>
              Register
            </h3>
            <hr />
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="exampleEmail"
                >
                  Email
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="email"
                  placeholder="example@company.com"
                  type="email"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="examplePassword"
                >
                  Password
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="password"
                  placeholder="password"
                  type="password"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="exampleFirstName"
                >
                  First Name
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="firstname"
                  placeholder="John"
                  type="firstname"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="exampleLastName"
                >
                  Last Name
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="lastname"
                  placeholder="Doe"
                  type="lastname"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="exampleCountry"
                >
                  Country
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="country"
                  placeholder=""
                  type="country"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="exampleState"
                >
                  State
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="state"
                  placeholder=""
                  type="state"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="exampleCity"
                >
                  City
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="city"
                  placeholder=""
                  type="city"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <div
                class="form-group"
              >
                <label
                  class=""
                  for="exampleAddress"
                >
                  Address
                </label>
                <input
                  aria-invalid="true"
                  class="is-invalid form-control"
                  name="address"
                  placeholder=""
                  type="address"
                  value=""
                />
                <div
                  class="invalid-feedback"
                />
              </div>
              <button
                class="btn btn-secondary"
                type="button"
              >
                Submit
              </button>
            </form>
            <h6>
              <br />
              Already registered? login
            </h6>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
